src/pages/home/TaskHeaderView.js

```diff
@@ -1,18 +1,72 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import compose from '../../libs/compose';
+import styles from '../../styles/styles';
 import reportPropTypes from '../reportPropTypes';
 import MenuItemWithTopDescription from '../../components/MenuItemWithTopDescription';
+import TaskSelectorLink from '../../components/TaskSelectorLink';
 import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
+import * as TaskUtils from '../../libs/actions/Task';
+import ONYXKEYS from '../../ONYXKEYS';
+import withLocalize from '../../components/withLocalize';
 
 const propTypes = {
     /** The report currently being looked at */
     report: reportPropTypes.isRequired,
+
+    /** All of the personal details for everyone */
+    personalDetails: PropTypes.objectOf(
+        PropTypes.shape({
+            /** Display name of the person */
+            displayName: PropTypes.string,
+
+            /** Avatar URL of the person */
+            avatar: PropTypes.string,
+
+            /** Login of the person */
+            login: PropTypes.string,
+        }),
+    ),
+};
+
+const defaultProps = {
+    personalDetails: {},
 };
 
 function TaskHeaderView(props) {
+    const [assignee, setAssignee] = useState('');
+
+    useEffect(() => {
+        TaskUtils.clearOutTaskInfo();
+        TaskUtils.setTaskReport(props.report);
+        if (!props.report.assignee) {
+            return;
+        }
+        if (props.report.assignee) {
+            const assigneeDetails = lodashGet(props.personalDetails, props.report.assignee);
+            const displayDetails = TaskUtils.constructAssignee(assigneeDetails);
+            setAssignee(displayDetails);
+        }
```

Comment:
We don't need the second if right? If we return early everything after should have an assignee

Reply:
True! Good catch